src/modules/base/Module.sol

```diff
@@ -60,22 +61,43 @@ abstract contract Module is IModule, Initializable, ContextUpgradeable {
 
     /// @notice Modifier to guarantee function is only callable by addresses
     ///         authorized via Orchestrator.
-    modifier onlyAuthorized() {
-        IAuthorizer authorizer = __Module_orchestrator.authorizer();
-        if (!authorizer.isAuthorized(_msgSender())) {
+    modifier onlyOrchestratorOwner() {
+        IAuthorizer authorizer =
+            IAuthorizer(address(__Module_orchestrator.authorizer()));
+
+        bytes32 ownerRole = authorizer.getOwnerRole();
+
+        if (!authorizer.hasRole(ownerRole, _msgSender())) {
             revert Module__CallerNotAuthorized();
         }
         _;
     }
 
     /// @notice Modifier to guarantee function is only callable by either
     ///         addresses authorized via Orchestrator or the Orchestrator's manager.
-    modifier onlyAuthorizedOrManager() {
-        IAuthorizer authorizer = __Module_orchestrator.authorizer();
+    modifier onlyOrchestratorOwnerOrManager() {
+        IAuthorizer authorizer =
+            IAuthorizer(address(__Module_orchestrator.authorizer()));
+
+        bytes32 ownerRole = authorizer.getOwnerRole();
+        bytes32 managerRole = authorizer.getManagerRole();
+
+        if (
+            !authorizer.hasRole(ownerRole, _msgSender())
+                && !authorizer.hasRole(managerRole, _msgSender())
+        ) {
+            revert Module__CallerNotAuthorized();
+        }
+        _;
+    }
+
+    /// @notice Modifier to guarantee function is only callable by addresses that hold a specific module-assigned role.
+    modifier onlyModuleRole(uint8 roleId) {
         if (
-            !authorizer.isAuthorized(_msgSender())
-                && __Module_orchestrator.manager() != _msgSender()
+            !IAuthorizer(address(__Module_orchestrator.authorizer()))
+                .isAuthorized(roleId, _msgSender())
         ) {
+            //revert Module__BountyManager__OnlyRole(roleId, address(this));
             revert Module__CallerNotAuthorized();
```

Comment:
hm maybe a bit more info on why the call exactly didnt run through makes sense
Also we with this its possible to check for exact role configurations in functions